src/Compilers/CSharp/Portable/FlowAnalysis/NullableWalker.cs

```diff
@@ -1641,25 +1662,70 @@ private static BoundExpression RemoveImplicitConversions(BoundExpression expr)
                 return Result.Unset;
             }
-
-            HashSet<DiagnosticInfo> useSiteDiagnostics = null;
-            TypeSymbol sourceType = valueResult.Type?.TypeSymbol;
-            var conversion = (sourceType is null) ?
-                _conversions.ClassifyImplicitConversionFromExpression(value, targetType, ref useSiteDiagnostics) :
-                _conversions.ClassifyImplicitConversionFromType(sourceType, targetType, ref useSiteDiagnostics);
+            var type = sourceType?.TypeSymbol;
+            var conversion = GenerateConversion(_conversions, sourceExpression, type, destinationType);
             if (!conversion.Exists)
             {
-                ReportStaticNullCheckingDiagnostics(ErrorCode.WRN_NullabilityMismatchInAssignment, value.Syntax, GetTypeAsDiagnosticArgument(valueResult.Type?.TypeSymbol), targetType);
+                Debug.Assert(GenerateConversion(_conversions.WithNullability(false), sourceExpression, type, destinationType).Exists);
+                ReportStaticNullCheckingDiagnostics(ErrorCode.WRN_NullabilityMismatchInAssignment, sourceExpression.Syntax, GetTypeAsDiagnosticArgument(type), destinationType);
             }
 
-            return InferResultNullability(value, conversion, targetType, valueResult.Type);
+            return InferResultNullability(sourceExpression, conversion, destinationType, sourceType);
+        }
+
+        // See Binder.BindNullCoalescingOperator for initial binding.
+        private bool GenerateConversionForConditionalOperator(BoundExpression sourceExpression, TypeSymbol sourceType, TypeSymbol destinationType, bool reportMismatch)
+        {
+            var conversion = GenerateConversion(_conversions, sourceExpression, sourceType, destinationType);
+            bool canConvert = conversion.Exists;
+            if (!canConvert && reportMismatch)
+            {
+                Debug.Assert(GenerateConversion(_conversions.WithNullability(false), sourceExpression, sourceType, destinationType).Exists);
+                ReportStaticNullCheckingDiagnostics(ErrorCode.WRN_NullabilityMismatchInAssignment, sourceExpression.Syntax, GetTypeAsDiagnosticArgument(sourceType), destinationType);
+            }
+            return canConvert;
+        }
+
+        private static Conversion GenerateConversion(Conversions conversions, BoundExpression sourceExpression, TypeSymbol sourceType, TypeSymbol destinationType)
+        {
+            HashSet<DiagnosticInfo> useSiteDiagnostics = null;
+            return UseExpressionForConversion(sourceExpression) ?
+                conversions.ClassifyImplicitConversionFromExpression(sourceExpression, destinationType, ref useSiteDiagnostics) :
+                conversions.ClassifyImplicitConversionFromType(sourceType, destinationType, ref useSiteDiagnostics);
+        }
+
+        // See ConversionsBase.ClassifyImplicitBuiltInConversionFromExpression.
+        private static bool UseExpressionForConversion(BoundExpression value)
```

Comment:
Consider adding a comment to the original methods, since it looks like they need to be kept in-sync.

Reply:
Not sure what was the outcome.

---
In reply to: [173348422](https://github.com/dotnet/roslyn/pull/25358#discussion_r173348422) [](ancestors = 173348422)

Reply:
`UseExpressionForConversion` was simplified so it no longer needs to stay in sync with `ConversionsBase` method.